### ISSUE TITLE
uint8 type

### ISSUE THREAD

**Issue by MBkkt** [ashvardanian/SimSIMD#134]
**not bit**

If we have vector of floats with only non-negative (or non-positive) values.
It's better to convert it to uint8 instead of int8

Examples under Apache 2.0:
https://github.com/google-research/google-research/tree/master/scann/scann/distance_measures/one_to_one
https://github.com/ydb-platform/ydb/tree/main/library/cpp/dot_product
https://github.com/ydb-platform/ydb/tree/main/library/cpp/l1_distance
https://github.com/ydb-platform/ydb/tree/main/library/cpp/l2_distance

**Comment by ashvardanian** [comment]
This sounds potentially relevant, but somewhat low-priority and significantly extends our surface area for tests and benchmarks. Let's wait until there is more demand for it until we introduce a new type 🤗 

PS: `int4_t` might be more interesting and relevant for SIMD ;)

**Comment by ashvardanian** [comment]
@MBkkt, what do you think about `fp8`? Are you looking to use SimSIMD in YDB? Having more companies participating in the R&D would make supporting more backends and numeric types much easier.

**Comment by MBkkt** [comment]
@ashvardanian 

> what do you think about fp8? 

I didn't see fp8, somewhere except faiss maybe.
But ofc more supported types is good.

> Are you looking to use SimSIMD in YDB?

Unfortunately no, using external library in this project is quite complicated.
So I just looked for good open source solutions to understand how to create a good solution for us.
Also we already have some of necessary functions.

I just wrote bunch of issue that I found, ofc they're are IMO, so feel free to close.

Also in my free time I'm a little benchmarked clang-18 with march=haswell/rocketlake and Ofast with your library and few other implementations (scann, ydb internals library), and found that compiler optimization is comparable in such case with manually written simd code (but types are important, int16 for int8, etc).

So I thought that maybe if we need such solution now, best what we can do, is write simple and correct code, and make dynamic dispatch.
For an example pg_vector doesn't use manually written simd
